fdbclient/NativeAPI.actor.cpp

```diff
@@ -3100,6 +3111,51 @@ ACTOR static Future<Void> commitDummyTransaction( Database cx, KeyRange range, T
 	}
 }
 
+ACTOR static Future<Optional<CommitID>> getCommitVersion(Database cx, TransactionOptions options, Version version,
+                                                         UID txID) {
+	state Transaction tr(cx);
+	state Key key = keyForTransaction(version, txID);
+	state int retries = 0;
+	state Future<Optional<Value>> minVersion;
+	state Future<Optional<Value>> versionStamp;
+	state Error err;
+	loop {
+		try {
+			tr.options = options;
+			tr.setOption(FDBTransactionOptions::TRANSACTION_IS_IDEMPOTENT);
+			versionStamp = tr.get(key);
+			minVersion = tr.get(transactionIDMinVersion);
+			wait(success(versionStamp) && success(minVersion));
+			if (minVersion.get().present()) {
+				auto v = decodeMinTxVersionValue(minVersion.get().get());
+				if (v > version) {
+					err = commit_unknown_result();
+					break;
+				}
+			}
+			if (versionStamp.get().present()) {
+				if (versionStamp.get().get().size() < 10) {
+					err = transaction_aborted();
```

Comment:
Is `transaction_aborted` an error that represents an unknown state?

Reply:
The idea here (and this is currently not correct due to a bug that you found below) is that this value is either not set or it is set to a versionstamp. A version stamp needs to be 10 bytes long. Therefore, this probably should throw a `transaction_aborted` - maybe even `internal_error` (as this typically would indicate that there's bug somewhere).